mypy/checkmember.py

```diff
@@ -945,7 +951,12 @@ def analyze_class_attribute_access(
             #         x: T
             #     C.x  # Error, ambiguous access
             #     C[int].x  # Also an error, since C[int] is same as C at runtime
-            if isinstance(t, TypeVarType) or has_type_vars(t):
+            # Exception is Self type wrapped in ClassVar, that is safe.
+            if node.node.info.self_type is not None and node.node.is_classvar:
+                exclude = {node.node.info.self_type.id}
+            else:
+                exclude = set()
```

Comment:
Optional optimization: this code path may be somewhat hot? Replacing a set with zero or one items with an optional type would be faster. You can use `!=` instead of ` not in`, I think.